miden-tx/src/host/mod.rs

```diff
@@ -225,6 +239,20 @@ impl<A: AdviceProvider> TransactionHost<A> {
         self.account_delta.vault_tracker().remove_asset(asset);
         Ok(())
     }
+
+    // HELPER FUNCTIONS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns the ID of the consumed note being executed.
+    fn get_current_note_id<S: ProcessState>(process: &S) -> Result<Option<NoteId>, ExecutionError> {
+        let note_address_felt = process
+            .get_mem_value(process.ctx(), CURRENT_CONSUMED_NOTE_PTR)
+            .expect("current consumed note pointer invalid")[0];
+        let note_address: u32 = note_address_felt
+            .try_into()
+            .map_err(|_| ExecutionError::MemoryAddressOutOfBounds(note_address_felt.as_int()))?;
+        Ok(process.get_mem_value(process.ctx(), note_address).map(NoteId::from))
+    }
```

Comment:
In case we haven't start executing any note, `expect` will panic. I wanted to return an `ExecutionError`, but looks like we don't have an error for the cases when the memory segment wasn't initialized. I can implement it, but I'm not sure that it is worth it since it is used only here.